packages/block-library/src/gallery/test/index.native.js

```diff
@@ -96,9 +104,44 @@ describe( 'Gallery block', () => {
 		expect( getByText( 'WordPress Media Library' ) ).toBeVisible();
 	} );
 
+	it( 'displays correct media options picker', async () => {
```

Comment:
Checking the media options displayed in the picker wasn't being tested due to an unrelated issue (https://github.com/WordPress/gutenberg/issues/38444). So I decided to split the test to check:
1. The correct media options _(here in this test)_
2. That block remains selected after dismissing the picker _(skipped until the issue is solved)_